src/GameData.ts

```diff
@@ -0,0 +1,320 @@
+            return `There are currently ${this.itemData.length} items in my lookup data!`;
+        }
+
+        let searchResult: SearchType[];
+        if (search.match(/^\d+$/)) {
+            searchResult = this.itemData.map(i => ({ item: i, score: levenshteinDistance(search, i.id.toString()) }));
+        } else {
+            searchResult = this.itemData.map(i => ({ item: i, score: levenshteinDistance(search, i.name.toLowerCase()) }));
+        }
+
+        searchResult = searchResult.sort((a, b) => this.sortSearch(search, a, b)).slice(0, this.sharedSettings.lookup.maxGuessCount);
+
+        // no exact match, so give alternatives
+        if (searchResult[0].score !== 0) {
+            if (searchResult.length > 0) {
+                let response = "Too many results returned for that search, did you mean one of the options below?\n```";
+                response += searchResult.map(x => `${x.item.name} (Alternate terms: ${x.item.id})`).join("\n");
+                response += "```";
+                return response;
+            } else {
+                return "Unable to find any data for that search term.";
+            }
+        }
+
+        return searchResult.map(x => x.item as ItemData)
+            .map(x => ({
+                id: x.id,
+                name: x.name,
+                combineCost: x.price,
+                cost: x.priceTotal,
+                from: x.from.length > 0 ? x.from : null,
+                to: x.to.length > 0 ? x.to : null,
+            })).slice(0, 1);
+    }
+
+    public findPerk(search: string): string | any[] {
+        if (!search) {
+            return `There are currently ${this.perkData.length} perks in my lookup data!`;
+        }
+
+        let searchResult: SearchType[];
+        if (search.match(/^\d+$/)) {
+            searchResult = this.perkData.map(i => ({ item: i, score: levenshteinDistance(search, i.id.toString()) }));
+        } else {
+            searchResult = this.perkData.map(i => ({ item: i, score: levenshteinDistance(search, i.name.toLowerCase()) }));
+        }
+
+        searchResult = searchResult.sort((a, b) => this.sortSearch(search, a, b)).slice(0, this.sharedSettings.lookup.maxGuessCount);
+
+        // no exact match, so give alternatives
+        if (searchResult[0].score !== 0) {
+            if (searchResult.length > 0) {
+                let response = "Too many results returned for that search, did you mean one of the options below?\n```";
+                response += searchResult.map(x => `${x.item.name} (Alternate terms: ${x.item.id})`).join("\n");
+                response += "```";
+                return response;
+            } else {
+                return "Unable to find any data for that search term.";
+            }
+        }
+
+        return searchResult.map(x => x.item as PerkData)
+            .map(x => ({
+                id: x.id,
+                name: x.name,
+                endOfGameStatDescs: x.endOfGameStatDescs,
+            })).slice(0, 1);
```

Comment:
Why the `slice` call? Now you're returning an array with a single element instead of an object. That just causes extra `[]`s in the response.